### ISSUE TITLE
[css-break-3] rules for direction to use when slicing inline borders for box-decoration-break:slice are unclear

### ISSUE THREAD

**Issue by dbaron** [w3c/csswg-drafts#4989]
The [rules for joining boxes with `box-decoration-break: slice`](https://drafts.csswg.org/css-break-3/#joining-boxes) currently say:

> **For boxes broken across lines**
> First, fragments on the same line are connected in visual order. Then, fragments on subsequent lines are ordered according to the element’s [inline base direction](https://drafts.csswg.org/css-writing-modes-4/#inline-base-direction) and aligned on the element’s dominant baseline. For example, in a left-to-right containing block (direction is ltr), the first fragment is the leftmost fragment on the first line and fragments from subsequent lines are put to the right of it. In a right-to-left containing block, the first fragment is the rightmost on the first line and subsequent fragments are put to the left of it. 

The normative text seems to say that it's the inline element's *own* direction, but the examples both refer to the *containing block's* direction.  It should be clear which one is used.

(I got here as part of filing [Mozilla bug 1632228](https://bugzilla.mozilla.org/show_bug.cgi?id=1632228).)

**Comment by fantasai** [comment]
Good catch. I think the better behavior would be to use the `direction` of the element (rather than the containing block) when joining fragments split across lines. Then, for example, if you have a gradient from the start to the end of the box, it will increase in parallel with progress through the text.

**Comment by r12a** [comment]
(Generally, i think it would be helpful to have more diagrams here – at least one for inline text broken across lines.)

I wanted to check what happens when a line break occurs inside a LTR sequence of words that is embedded in a RTL paragraph. Here is an example from https://r12a.github.io/scripts/tutorial/part5#latin-in-rtl

![Screenshot 2020-10-27 at 11 37 05](https://user-images.githubusercontent.com/4839211/97296685-e4301380-1848-11eb-9ca3-5b189b7c9a5b.png)

Note how 'Unicode Conference' (which would have 'Unicode' on the left normally, remains on the first line when rearranged around the line break.  This seems to suggest to me that it's the parent's directionality that needs to be taken into account, rather than the direction of the inline element.

I'd personally expect (though we may want to check with native speakers) to see

![Screenshot 2020-10-27 at 11 44 31](https://user-images.githubusercontent.com/4839211/97297412-ed6db000-1849-11eb-97ea-841103c2e600.png)

become

![Screenshot 2020-10-27 at 11 37 05](https://user-images.githubusercontent.com/4839211/97297438-f6f71800-1849-11eb-8f76-2a59d4943155.png)

even though the 'U' was originally against a vertical line.

Do we need to change anything here to make this work right?

**Comment by r12a** [comment]
If you need it for testing or creating examples, i can provide the text of the above.

**Comment by frivoal** [comment]
@r12a my intuition about what should happen to a border in this situation matches what you described, so so far I'm happy. 

However, I'm terribly confused as to what ought to happen if the inline element had a background. Say that before fragmentation, it had a background gradient going from cyan under the U of Unicode to pink under the last e of Conference. What would you expect to see what it's broken into two lines with `box-decoration-break: slice`? What if the background was a long rightwards arrow, or something photo realistic, does that make a difference?

**Comment by r12a** [comment]
@frivoal, indeed. My picture was just an example, and i was expecting that the question would get more complicated when talking about backgrounds.  I don't know the answers, and could see a case for the directionality of the backgrounds following the text but also for the backgrounds to follow the overall direction of the paragraph – depending on the nature of the background.  Perhaps the properties need to make that clear??

**Comment by xfq** [comment]
It feels a bit like the problem here (but not exactly the same, because there is no fragmentation or line break in the html-bidi example): https://www.w3.org/TR/html-bidi/#image-flip

I wonder if it's possible to solve this issue by recommending the author to [specify a directionality](https://drafts.csswg.org/css-images-4/#bidi-images) for the background gradient/image?
